5calls/app/src/main/java/org/a5calls/android/a5calls/controller/LocationActivity.java

```diff
@@ -137,20 +136,9 @@ private void submitZip() {
     }
 
     private void returnToMain() {
-        // Make sure we're still alive
-        if (isFinishing() || isDestroyed()) {
-            return;
-        }
-
-        // If we came from MainActivity and return with another Intent, it will create a deep stack
-        // of activities!
-        if (mFromMain) {
-            onBackPressed();
-        } else {
-            Intent intent = new Intent(this, MainActivity.class);
-            startActivity(intent);
-            finish();
-        }
+        Intent intent = new Intent(this, MainActivity.class);
```

Comment:
Oh, I see your explanation above. Can we move this into a different PR to keep it separate from all this functionality? I'd like to experiment and see if there's a way to make it work without multiple layers of activities.

One option may be that we "startActivityForResult" from MainActivity when the Location menu option is clicked, and then we can "onActivityResult" instead of "onBackPressed". I'd need to look into it more, though.